src/redisjson.rs

```diff
@@ -402,16 +412,22 @@ pub mod type_methods {
 
     #[allow(non_snake_case, unused)]
     pub unsafe extern "C" fn free(value: *mut c_void) {
-        Box::from_raw(value as *mut RedisJSON);
+        let json = value as *mut RedisJSON;
+        if let Some(index) = &(*json).value_index {
+            index::remove_document(&index.key, &index.index);
+        }
+
+        Box::from_raw(json);
```

Comment:
This code can be cleaned up a bit by boxing first instead of using the raw pointer (and adding a comment):
```
        let json = value as *mut RedisJSON;

        // Take ownership of the data from Redis (causing it to be dropped when we return)
        let json = Box::from_raw(json);

        if let Some(index) = &json.value_index {
            index::remove_document(&index.key, &index.index);
        }
```

Reply:
:+1: